processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -1830,6 +1869,18 @@ public List<SubclassMapping> getSubclassMappings() {
         return subclassMappings;
     }
 
+    public String getFinalizedResultName() {
+        return getResultName() + "Result";
+    }
```

Comment:
I would initialize the `finalizedResultName` in the constructor. Otherwise, this can potentially lead to compiler errors due to the name already being used.

Reply:
Yes you're right. I had a bad feeling putting this in the getter.